WordPress/Classes/ViewRelated/System/WPTabBarController.m

```diff
@@ -87,6 +101,10 @@ - (instancetype)init
                                                  selector:@selector(defaultAccountDidChange:)
                                                      name:WPAccountDefaultWordPressComAccountChangedNotification
                                                    object:nil];
+
+        // Watch for application badge number changes
+        NSString *badgeKeyPath = NSStringFromSelector(@selector(applicationIconBadgeNumber));
```

Comment:
I know this is from the original code but, since you're working on it...

We have three places in our code where we repeat something similar to this.  We can improve this code by replacing:

```objective-c
NSStringFromSelector(@selector(applicationIconBadgeNumber))
```

with a call to a method like this one:

```objective-c
- (NSString*)applicationIconBadgeNumberKeypath
{
    static NSString* applicationBadgeCountKeypath = @"applicationIconBadgeNumber";

    NSAssert([[UIApplication sharedApplication] respondsToSelector:@selector(applicationBadgeCountKeypath)]);
    return applicationBadgeCountKeypath;
}
```

Advantages:

- Not need to call `NSStringFromSelector(@selector(...))` each time.  Keep in mind using `@selector()` does not ensure the method actually exists.
- Makes sure through the assert that we don't change the method name by mistake.  Currently there's no protection for this.
- Reduces redundancy.